templates/_helpers.tpl

```diff
@@ -470,6 +470,31 @@ Sets extra injector service annotations
   {{- end }}
 {{- end -}}
 
+{{/*
+securityContext for the injector pod level.
+*/}}
+{{- define "injector.securityContext.pod" -}}
+  {{- if or (.Values.injector.uid) (.Values.injector.gid) }}
+      securityContext:
+        runAsNonRoot: true
+        runAsGroup: {{ .Values.injector.gid | default 1000 }}
+        runAsUser: {{ .Values.injector.uid | default 100 }}
+  {{- else }}
```

Comment:
Might as well check if this securityContext is set, similar to `securityContext.container`.

```suggestion
  {{- else if .Values.injector.securityContext.pod }}
```